Compile warnings in tst_interops5.c

Some compile warnings that will be easy to fix:

```
tst_interops5.c: In function ‘main’:
tst_interops5.c:239:40: warning: passing argument 3 of ‘nc_inq_grps’ from incompatible pointer type [-Wincompatible-pointer-types]
       if (nc_inq_grps(ncid, &ngrps_in, &grpid)) ERR;
                                        ^
In file included from ../include/nc_tests.h:22:0,
                 from tst_interops5.c:8:
../include/netcdf.h:524:1: note: expected ‘int *’ but argument is of type ‘hid_t * {aka long int *}’
 nc_inq_grps(int ncid, int *numgrps, int *ncids);
 ^~~~~~~~~~~
tst_interops5.c:203:12: warning: unused variable ‘name’ [-Wunused-variable]
       char name[MAX_NAME + 1], name_in[MAX_NAME + 1];
            ^~~~
tst_interops5.c:202:14: warning: unused variable ‘namelen’ [-Wunused-variable]
       size_t namelen = MAX_NAME;
              ^~~~~~~
tst_interops5.c:201:14: warning: unused variable ‘cd_nelems’ [-Wunused-variable]
       size_t cd_nelems = NUM_CD_ELEM;
              ^~~~~~~~~
tst_interops5.c:200:51: warning: unused variable ‘filter_config’ [-Wunused-variable]
       unsigned int flags, cd_values[NUM_CD_ELEM], filter_config;
                                                   ^~~~~~~~~~~~~
tst_interops5.c:200:27: warning: unused variable ‘cd_values’ [-Wunused-variable]
       unsigned int flags, cd_values[NUM_CD_ELEM], filter_config;
                           ^~~~~~~~~
tst_interops5.c:200:20: warning: unused variable ‘flags’ [-Wunused-variable]
       unsigned int flags, cd_values[NUM_CD_ELEM], filter_config;
                    ^~~~~
tst_interops5.c:198:11: warning: unused variable ‘num_filters’ [-Wunused-variable]
       int num_filters;
           ^~~~~~~~~~~
tst_interops5.c:197:20: warning: unused variable ‘filter’ [-Wunused-variable]
       H5Z_filter_t filter;
                    ^~~~~~
```

--- a/nc_test4/tst_interops5.c
+++ b/nc_test4/tst_interops5.c
@@ -194,14 +194,10 @@ main(int argc, char **argv)
       hid_t fileid, grpid, spaceid, datasetid;
       int data_out[DIM1_LEN], data_in[DIM1_LEN];
       hsize_t dims[1] = {DIM1_LEN};
-      H5Z_filter_t filter;
-      int num_filters;
       hid_t propid;
-      unsigned int flags, cd_values[NUM_CD_ELEM], filter_config;
-      size_t cd_nelems = NUM_CD_ELEM;
-      size_t namelen = MAX_NAME;
-      char name[MAX_NAME + 1], name_in[MAX_NAME + 1];
+      char name_in[MAX_NAME + 1];
       int ncid, ndims_in, nvars_in, ngatts_in, unlimdimid_in, ngrps_in;
+      int nc_grpid;
       int dimid_in[1], natts_in;
 
       nc_type xtype_in;
@@ -236,19 +232,19 @@ main(int argc, char **argv)
       if (nc_open(FILE_NAME, NC_NOWRITE, &ncid)) ERR;
       if (nc_inq(ncid, &ndims_in, &nvars_in, &ngatts_in, &unlimdimid_in)) ERR;
       if (ndims_in != 0 || nvars_in != 0 || ngatts_in != 0 || unlimdimid_in != -1) ERR;
-      if (nc_inq_grps(ncid, &ngrps_in, &grpid)) ERR;
+      if (nc_inq_grps(ncid, &ngrps_in, &nc_grpid)) ERR;
       if (ngrps_in != 1) ERR;
-      if (nc_inq(grpid, &ndims_in, &nvars_in, &ngatts_in, &unlimdimid_in)) ERR;
+      if (nc_inq(nc_grpid, &ndims_in, &nvars_in, &ngatts_in, &unlimdimid_in)) ERR;
       if (ndims_in != 1 || nvars_in != 1 || ngatts_in != 0 || unlimdimid_in != -1) ERR;
 
       /* Check the variable. */
-      if (nc_inq_var(grpid, 0, name_in, &xtype_in, &ndims_in, dimid_in,
+      if (nc_inq_var(nc_grpid, 0, name_in, &xtype_in, &ndims_in, dimid_in,
       		     &natts_in)) ERR;
       if (strcmp(name_in, BATTLE_RECORD) || xtype_in != NC_INT || ndims_in != 1 ||
       	  dimid_in[0] != 0 || natts_in != 0) ERR;
 
       /* Check the data. */
-      if (nc_get_var(grpid, 0, data_in)) ERR;
+      if (nc_get_var(nc_grpid, 0, data_in)) ERR;
       for (i = 0; i < DIM1_LEN; i++)
 	 if (data_in[i] != data_out[i]) ERR;
 

